File: solutions/fortran/leap/1/leap.f90
module leap
  implicit none

contains

  logical function divides(n, k)
    integer :: n
    integer :: k

    divides = mod(n, k) .eq. 0
  end function

  logical function is_leap_year(year)
    integer :: year

    is_leap_year = divides(year, 400) .or. (divides(year, 4) .and. (.not. divides(year, 100)))
  end function

end module

File: solutions/clojure/leap/1/src/leap.clj
(ns leap)

(defn leap-year?
  "Returns true if the given year is a leap year;
  otherwise, it returns false."
  [year]
  (or
    (== 0 (mod year 400))
    (and (== 0 (mod year 4)) (not (== 0 (mod year 100)))
  )))
